src/services/contentful.js

```diff
@@ -82,4 +89,12 @@ async function getStories() {
   return null;
 }
 
-export default { getEntries: fetchContentfulEntry, getMetadata, getStories };
\ No newline at end of file
+async function getTexts() {
```

Comment:
I would do the same thing as we do with the metadata. Instead of fetching all the text entries at once would be preferable to fetch them once they're needed. Passing some filter params will do the trick:
```js
async function getTexts(slug) {
  const data = await fetchContentfulEntry({ contentType: 'texts', filterField:'view', filterValue: slug });
  if (data && data.items && data.items.length > 0) {
    return parseTexts(data.items);
  }
  return null;
}
```